policy/modules/services/container.if

```diff
@@ -0,0 +1,840 @@
+	filetrans_pattern($2, container_data_home_t, container_ro_file_t, dir, "overlay")
+	filetrans_pattern($2, container_data_home_t, container_ro_file_t, dir, "overlay-images")
+	filetrans_pattern($2, container_data_home_t, container_ro_file_t, dir, "overlay-layers")
+	filetrans_pattern($2, container_data_home_t, container_ro_file_t, dir, "overlay2")
+	filetrans_pattern($2, container_data_home_t, container_ro_file_t, dir, "overlay2-images")
+	filetrans_pattern($2, container_data_home_t, container_ro_file_t, dir, "overlay2-layers")
+	filetrans_pattern($2, container_data_home_t, container_file_t, dir, "volumes")
+
+	optional_policy(`
+		dbus_spec_session_bus_client($1, podman_user_t)
+
+		systemd_spec_user_read_state($1, container_engine_user_domain)
+		systemd_spec_user_system_start($1, container_engine_user_domain)
+		systemd_spec_user_system_stop($1, container_engine_user_domain)
+		systemd_spec_user_system_status($1, container_engine_user_domain)
+		systemd_spec_user_dbus_chat($1, container_engine_user_domain)
+
+		systemd_user_app_status($1, container_user_domain)
+	')
+')
+
+########################################
+## <summary>
+##	Execute generic container engines in the
+##	container engine domain.
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed to transition.
+## </summary>
+## </param>
+#
+interface(`container_domtrans_generic_engine',`
+	gen_require(`
+		type container_engine_t, container_engine_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, container_engine_exec_t, container_engine_t)
+')
+
+########################################
+## <summary>
+##	Execute generic container engines in the
+##	container engine domain, and allow the
+##	specified role the container domain.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition
+##	</summary>
+## </param>
+## <param name="role">
+##	<summary>
+##	The role to be allowed the container domain.
+##	</summary>
+## </param>
+#
+interface(`container_run_generic_engine',`
+	gen_require(`
+		attribute_role container_roles;
+	')
+
+	container_domtrans_generic_engine($1)
+	roleattribute $2 container_roles;
+')
+
+########################################
+## <summary>
+##	Execute docker in the docker domain.
+## </summary>
+## <param name="domain">
+## <summary>
+##	Domain allowed to transition.
+## </summary>
+## </param>
+#
+interface(`container_domtrans_docker',`
+	gen_require(`
+		type docker_t, docker_exec_t;
+	')
+
+	corecmd_search_bin($1)
+	domtrans_pattern($1, docker_exec_t, docker_t)
+')
+
+########################################
+## <summary>
+##	Execute docker in the docker domain,
+##	and allow the specified role the
+##	container domain.
+## </summary>
+## <param name="domain">
+##	<summary>
+##	Domain allowed to transition
+##	</summary>
+## </param>
+## <param name="role">
+##	<summary>
+##	The role to be allowed the container domain.
+##	</summary>
+## </param>
+#
+interface(`container_run_docker',`
```

Comment:
Does running the docker cli really need this? Or is this for a user domain running a docker daemon for itself?

Reply:
This is so that `sysadm_t` or `unconfined_t` can run the Docker CLI and talk to the Docker daemon. They are running in the same domain, but they could probably be split out into `dockerd_t` for the Docker daemon and `dockerc_t` for the Docker CLI.

I didn't think about the possibility of running the Docker daemon as a user. That is probably something that should also be accounted for when I move the Docker policy to its own module as previously requested.